Server/mods/deathmatch/logic/luadefs/CLuaResourceDefs.cpp

```diff
@@ -898,23 +898,33 @@ int CLuaResourceDefs::getResourceLoadTime(lua_State* luaVM)
     return 1;
 }
 
-int CLuaResourceDefs::getResourceName(lua_State* luaVM)
+std::string CLuaResourceDefs::GetResourceName(lua_State* luaVM, std::optional<CResource*> resourceElement)
 {
-    CResource* pResource;
+    if (!resourceElement)
+    {
+        CLuaMain*  localVM = m_pLuaManager->GetVirtualMachine(luaVM);
+        CResource* localResource;
 
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadUserData(pResource);
+        if (!localVM)
+            throw std::runtime_error("Couldn't find the virtual machine");
 
-    if (!argStream.HasErrors())
-    {
-        lua_pushstring(luaVM, pResource->GetName().c_str());
-        return 1;
+        localResource = localVM->GetResource();
+
+        if (!localResource)
+            throw std::runtime_error("Couldn't find the resource");
+
+        resourceElement = localResource;
     }
-    else
-        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
 
-    lua_pushboolean(luaVM, false);
-    return 1;
+    if (!(*resourceElement)->IsActive())
+        throw std::runtime_error("The resource isn't active");
+
+    std::string resourceName = (*resourceElement)->GetName();
+
+    if (resourceName.empty())
+        throw std::runtime_error("The resource name is empty");
```

Comment:
There's no need in this check because you simply cannot create a resource with empty name in MTA.

Reply:
Thank you.